tests/integration/regenerate.sh

```diff
@@ -39,30 +38,37 @@ fi
 
 export SCRIPT_DIR=$(get_script_dir)
 export PROJECT_ROOT=$(realpath "$SCRIPT_DIR/../..")
-export LOG_DIR=$PROJECT_ROOT/logs
+export LOG_DIR="$PROJECT_ROOT/logs"
 echo "Current working directory: $(pwd)"
 echo "SCRIPT_DIR: $SCRIPT_DIR"
 echo "PROJECT_ROOT: $PROJECT_ROOT"
 echo "LOG_DIR: $LOG_DIR"
 echo "LOG_TO_FILE: $LOG_TO_FILE"
 
 WORKSPACE_BASE=${WORKSPACE_BASE}/_test_workspace
-mkdir -p $WORKSPACE_BASE
-chmod -R 777 $WORKSPACE_BASE
-WORKSPACE_BASE=$(realpath $WORKSPACE_BASE)
+mkdir -p "$WORKSPACE_BASE"
+chmod -R 777 "$WORKSPACE_BASE"
+WORKSPACE_BASE=$(realpath "$WORKSPACE_BASE")
 
-WORKSPACE_MOUNT_PATH=${WORKSPACE_MOUNT_PATH}/_test_workspace
-mkdir -p $WORKSPACE_MOUNT_PATH
-chmod -R 777 $WORKSPACE_MOUNT_PATH
-WORKSPACE_MOUNT_PATH=$(realpath $WORKSPACE_MOUNT_PATH)
+if [ -z "$WORKSPACE_MOUNT_PATH" ]; then
+  WORKSPACE_MOUNT_PATH="$WORKSPACE_BASE"
+else
+  WORKSPACE_MOUNT_PATH="${WORKSPACE_MOUNT_PATH}/_test_workspace"
```

Comment:
Can you please explain this change: it seems we don't create a `_test_workspace` anymore in all cases? If we don't create a `_test_workspace`, then the script can't clean up what it did for a test, I think it currently - usually - does manage to delete its stuff.

Reply:
If you look at the full code block there, WORKSPACE_MOUNT_PATH in every case gets the `_test_workspace`.

Reply:
IF it is not set, it gets the WORKSPACE_BASE value (which has the _test_workspace already at that point), otherwise it gets the suffix added in line 56.